private/pkg/git/gitbranch/doc.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gitbranch implements a Ranger to range over branches
+// and commits in branches.
+package gitbranch
```

Comment:
Why a new package for this? Why not have `git.BranchIterator`?

Reply:
No strong reason, they're a lot of new types and I didn't want to pollute the `git` package. I don't feel strongly about it, so I can merge `gitbranch` and `gitobject` into there (and change the type names).